internal/api/token.go

```diff
@@ -122,27 +122,29 @@ func (a *API) ResourceOwnerPasswordGrant(ctx context.Context, w http.ResponseWri
 
 	grantParams.FillGrantParams(r)
 
-	if params.Email != "" {
+	switch {
+	case params.Email != "" && config.External.Email.Enabled:
 		provider = "email"
-		if !config.External.Email.Enabled {
-			return badRequestError("Email logins are disabled")
-		}
 		user, err = models.FindUserByEmailAndAudience(db, params.Email, aud)
-	} else if params.Phone != "" {
+
+	case params.Phone != "" && config.External.Phone.Enabled:
 		provider = "phone"
-		if !config.External.Phone.Enabled {
-			return badRequestError("Phone logins are disabled")
-		}
 		params.Phone = formatPhoneNumber(params.Phone)
 		user, err = models.FindUserByPhoneAndAudience(db, params.Phone, aud)
-	} else {
+
+	case params.Email != "":
+		return badRequestError("Email logins are disabled")
+
+	case params.Phone != "":
+		return badRequestError("Phone logins are disabled")
```

Comment:
i think it's more straightforward to check if `!config.External.Email.Enabled` and `!config.External.Phone.Enabled` here instead of whether the params passed are empty strings

Reply:
Yeah that makes sense, let's do that